nim/tests/test_routes.py

```diff
@@ -0,0 +1,110 @@
+import json
+from nim.app import app
+import pytest
+
+
+@pytest.fixture
+def client_app():
+    return app.test_client()
+
+
+def assert_new_game_success(passed_client, req):
+    """
+    Asserts post to /new is 200
+    & state is not empty
+    :param passed_client: test_client
+    :param req: json to post
+    :return: bool
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
+    assert response.status_code == 200 and not len(json.loads(response.data)['state']) == 0
+
+
+def assert_new_game_failed(passed_client, req):
+    """
+    Asserts post tp /new gets error code 400
+     & the string 'error' exists in the response
+    :param passed_client: test_client
+    :param req: json to post
+    :return: bool
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
+    assert response.status_code == 400 and b'error' in response.data
```

Comment:
We also want to make sure we're not returning a state.
And you may be able to do this instead: `and 'error' in json.loads(response.data)` so you don't have to use the `b` prefix.
And it would be nice to check that the error message isn't empty.